core/endpoint.go

```diff
@@ -221,6 +224,14 @@ func (endpoint *Endpoint) ValidateAndSetDefaults() error {
 	if _, contentTypeHeaderExists := endpoint.Headers[ContentTypeHeader]; !contentTypeHeaderExists && endpoint.GraphQL {
 		endpoint.Headers[ContentTypeHeader] = "application/json"
 	}
+
+	// Automatically add Origin header for websocket endpoints if there isn't one specified
+	if endpoint.Type() == EndpointTypeWS {
+		if _, originHeaderExists := endpoint.Headers[OriginHeader]; !originHeaderExists {
+			endpoint.Headers[OriginHeader] = Origin
+		}
+	}
```

Comment:
While I agree that adding support for passing the origin header is worth it, I think perhaps we shouldn't add it by default here but instead add a note in the docs mentioning that users _may_ have to specify an `Origin` header.

Reply:
@TwiN It seems as the code as in this PR ensures that the current behavior, which users might rely on, continues to functions as it has.  I'm unsure if ceasing the specify the Origin header will end up being a backwards incompatible change. What is the benefit of no longer specifying the Origin header on any websocket call unless it is explicitly specified?